lib/concurrent/atomic_reference.rb

```diff
@@ -0,0 +1,104 @@
+  #       CAtomicReference      0.040000   0.000000   0.040000 (  0.034012)
+  #       
+  #       *** Parallel updates ***
+  #                                 user     system      total        real
+  #       no lock               0.010000   0.000000   0.010000 (  0.009387)
+  #       mutex                 0.030000   0.010000   0.040000 (  0.032545)
+  #       MutexAtomicReference  0.830000   2.280000   3.110000 (  2.146622)
+  #       CAtomicReference      0.040000   0.000000   0.040000 (  0.038332)
+  #
+  #       Testing with jruby 1.9.3
+  #       
+  #       *** Sequential updates ***
+  #                                 user     system      total        real
+  #       no lock               0.170000   0.000000   0.170000 (  0.051000)
+  #       mutex                 0.370000   0.010000   0.380000 (  0.121000)
+  #       MutexAtomicReference  1.530000   0.020000   1.550000 (  0.471000)
+  #       JavaAtomicReference   0.370000   0.010000   0.380000 (  0.112000)
+  #       
+  #       *** Parallel updates ***
+  #                                 user     system      total        real
+  #       no lock               0.390000   0.000000   0.390000 (  0.105000)
+  #       mutex                 0.480000   0.040000   0.520000 (  0.145000)
+  #       MutexAtomicReference  1.600000   0.180000   1.780000 (  0.511000)
+  #       JavaAtomicReference   0.460000   0.010000   0.470000 (  0.131000)
+  #
+  #   @see http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html
+  #   @see http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/package-summary.html
+  class Concurrent::AtomicReference < Concurrent::JavaAtomicReference
+  end
+
+  # @!macro atomic_reference
+  # @deprecated Use Concurrent::JavaAtomicReference instead.
+  class Concurrent::JavaAtomic < Concurrent::JavaAtomicReference
+  end
+
+elsif defined? Concurrent::RbxAtomicReference
+
+  # @!macro atomic_reference
+  class Concurrent::AtomicReference < Concurrent::RbxAtomicReference
+  end
+
+  # @!macro atomic_reference
+  # @deprecated Use Concurrent::RbxAtomicReference instead.
+  class Concurrent::RbxAtomic < Concurrent::RbxAtomicReference
+  end
+
+elsif defined? Concurrent::CAtomicReference
+
+  # @!macro atomic_reference
+  class Concurrent::AtomicReference < Concurrent::CAtomicReference
+  end
+
+  # @!macro atomic_reference
+  # @deprecated Use Concurrent::CAtomicReference instead.
+  class Concurrent::CAtomic < Concurrent::CAtomicReference
+  end
+
+else
+
+  # @!macro atomic_reference
+  class Concurrent::AtomicReference < Concurrent::MutexAtomicReference
+  end
+end
+
+# @!macro atomic_reference
+# @deprecated Use Concurrent::MutexAtomicReference instead.
+class Concurrent::MutexAtomic < Concurrent::MutexAtomicReference
+end
+
+# @!macro atomic_reference
+# @deprecated Use Concurrent::AtomicReference instead.
+class Concurrent::Atomic < Concurrent::AtomicReference
+end
```

Comment:
Do we need aliases for the different implementations? In my mind they are not included in the public API. Just assignment `Atomic = AtomicReference` might be better since it clearly says what is the main name and the alias will also correctly report full name `AtomicReference`.